### ISSUE TITLE
useQuery wrapper calls API twice

### ISSUE THREAD

**Issue by pedropmedina** [ts-rest/ts-rest#300]
I'm not sure if anyone else is experiencing this issue, but I have tried several solution and nothing works. It seems as if the useQuery wrapper makes two calls over the network. This issue is NOT present when using react-query's native useQuery hook and calling the api with the ts-rest regular client as the queryFn. 

Here's what I'm seeing

 <img width="1429" alt="Screenshot 2023-07-11 at 12 07 08 AM" src="https://github.com/ts-rest/ts-rest/assets/25961906/1c677a13-23d9-45e4-b14a-52bdc4978abb">

 
 ```javascript
 export const api = initQueryClient(contract, {
  baseUrl: `${process.env.NEXT_PUBLIC_AUTH_API_URL}`,
  jsonQuery: true,
  baseHeaders: {},
  credentials: 'include',
  
  //  This makes two calls
    const { data, isInitialLoading } = api.application.findMany.useQuery(QUERY_KEY, {
    query: { take: 100 },
  });

// This one makes a single call
  const { data, isInitialLoading } = useQuery({
    queryKey: QUERY_KEY,
    queryFn: async () => {
      const res = await api.application.findMany.query({ query: { take: 100 } });
      return res.status === 200 ? res : null;
    },
  });
});
 ```

**Comment by danielpza** [comment]
perhaps this is caused by react strict mode on development? https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development

**Comment by pedropmedina** [comment]
You are right @danielpza . The issue was strict mode. I'm using NextJs and turning it off got rid of the problem. However, doing so is not recommended. I'm not sure why this happens with the ts-rest wrapper and not when using react-query's native useQuery? Any thoughts?

**Comment by kurdin** [comment]
@pedropmedina 
maybe this can help to explain it
https://github.com/facebook/react/issues/24502#issuecomment-1120856043

**Comment by pedropmedina** [comment]
I get what React is doing with strict mode @kurdin. I just don't understand why the ts-rest wrapper around react-query's userQuery is causing double fetching when this issue was already addressed by react-query's useQuery itself. In order words, using native useQuery triggers a single request while ts-rest's implementation triggers two. I think there's something in that helper that must be improved to avoid this behavior. At the end of the day, aren't we just calling native useQuery under the hood?

**Comment by kurdin** [comment]
You right @pedropmedina. I'll try to use ts-rest with react query to investigate this behaviour.

**Comment by michaelangelo-io** [comment]
@pedropmedina given the comments on this, https://github.com/TanStack/query/issues/3492
do you see if the first request is aborted or not?

**Comment by pedropmedina** [comment]
That's what I was expecting @michaelangelo-io, however both request go through the network successfully as shown in the image above ^. The weird thing is that this only happens when using ts-rest wrapper. Native useQuery works as expected.

**Comment by Talent30** [comment]
@pedropmedina @kurdin  Hi, there. I am happy to have a look if you can provide a minimal repo for to reproduce the behaviour.
